snuba/query/parser/__init__.py

```diff
@@ -127,9 +134,25 @@ def build_selected_expressions(
 
     arrayjoin = body.get("arrayjoin")
     if arrayjoin:
-        array_join_expr: Optional[Expression] = parse_expression(body["arrayjoin"])
+        array_join_expr: Optional[Expression] = parse_expression(
```

Comment:
I noticed that you changed the main parser but did not add a query parsing test. Could you please add a test to `tests/query/parser/test_query` With this case? It would be a stronger coverage than the test_expression ones.